gapis/api/vulkan/frame_loop.go

```diff
@@ -380,108 +402,51 @@ func (f *frameLoop) Transform(ctx context.Context, cmdId api.CmdID, cmd api.Cmd,
 				f.buildStartEndStates(ctx, out.State())
 				f.detectChangedResources(ctx)
 
-				apiState := GetState(out.State())
-
-				stateBuilder := apiState.newStateBuilder(ctx, newTransformerOutput(out))
-				defer stateBuilder.ta.Dispose()
-
 				// Back up the resources that change in the loop (as indentified above)
 				if err := f.backupChangedResources(ctx, stateBuilder); err != nil {
 					log.E(ctx, "FrameLoop: Failed to backup changed resources: %v", err)
 					return
 				}
 
-				// Write out some custom bytecode for the start of the loop...
-				stateBuilder.write(stateBuilder.cb.Custom(func(ctx context.Context, s *api.GlobalState, b *builder.Builder) error {
-					f.loopCountPtr = b.AllocateMemory(4)
-					b.Push(value.S32(f.loopCount))
-					b.Store(f.loopCountPtr)
-					b.JumpLabel(uint32(0x1))
-					return nil
-				}))
-
 				// Notify the other transforms that we're about to emit the start of the loop.
 				out.NotifyPreLoop(ctx)
-
-				// Write out the command that started all of this loop work.
-				out.MutateAndWrite(ctx, f.capturedLoopCmdIds[0], f.capturedLoopCmds[0])
 			}
 
-			// Do mid-loop stuff.
-			{
-				cb := CommandBuilder{Thread: cmd.Thread(), Arena: out.State().Arena}
-
-				// Iterate through the loop contents, emitting instructions one by one.
-				for cmdIndex, cmd := range f.capturedLoopCmds {
-
-					switch cmd.(type) {
-					case *VkUnmapMemory:
-						vkCmd := cmd.(*VkUnmapMemory)
-						mem := vkCmd.Memory()
-						if _, ok := f.memoryToMap[mem]; !ok {
-							break
-						}
-
-						memObj := GetState(out.State()).DeviceMemories().Get(mem)
-						addr := memObj.MappedLocation().Address()
-						if addr == 0 {
-							break
-						}
-
-						// Only remember the first mapped target.
-						if _, ok := f.mappedAddress[addr]; !ok {
-							out.MutateAndWrite(ctx, api.CmdNoID, cb.Custom(func(ctx context.Context, s *api.GlobalState, b *builder.Builder) error {
-								target, err := b.GetMappedTarget(value.ObservedPointer(addr))
-								if err == nil {
-									f.mappedAddress[addr] = target
-								}
-								return err
-							}))
-						}
-					}
+			// Do first iteration of mid-loop stuff.
+			f.writeLoopContents(ctx, cmd, out)
 
-					// We've already processed the first loop command, so skip that one.
-					// All others get emitted.
-					if cmdIndex > 0 {
-						out.MutateAndWrite(ctx, f.capturedLoopCmdIds[cmdIndex], cmd)
-					}
-				}
-			}
-
-			// Do end loop stuff.
+			// Mark branch target for loop jump
 			{
-				apiState := GetState(out.State())
-
-				stateBuilder := apiState.newStateBuilder(ctx, newTransformerOutput(out))
-				defer stateBuilder.ta.Dispose()
-
-				// Write out the command that ended the loop. That one is outside of the captured commands since it's the one that called this code.
-				out.MutateAndWrite(ctx, cmdId, cmd)
-
-				// Notify the other transforms that we have just emitted the end of the loop.
-				out.NotifyPostLoop(ctx)
```

Comment:
In the new code, the `out.NotifyPostLoop(ctx)` is gone. This is need to notify other transform to clean up at the end of loop